lib/Conversion/QUIRToPulse/QUIRToPulse.cpp

```diff
@@ -111,26 +111,17 @@ void QUIRToPulsePass::runOnOperation() {
     callCircOp->erase();
   });
 
-  // erase the quir circuits
-  LLVM_DEBUG(llvm::dbgs() << "\nErasing quir circuits:\n");
-  for (auto *op : quirCircuitEraseList) {
-    LLVM_DEBUG(op->dump());
-    op->erase();
-  }
-
-  // erase quir barriers before erasing the operands
-  moduleOp->walk([&](mlir::quir::BarrierOp barrierOp) { barrierOp->erase(); });
-
-  // erase the quir circuit operands
-  LLVM_DEBUG(llvm::dbgs() << "\nErasing quir circuit operands:\n");
-  for (auto *op : quirCircuitOperandEraseList) {
-    LLVM_DEBUG(op->dump());
-    op->erase();
-  }
+  // erase circuit ops
+  moduleOp->walk([&](CircuitOp circOp) { circOp->erase(); });
 
-  // erase the rest of quir.declare_qubits (unused in the input program)
-  moduleOp->walk([&](mlir::quir::DeclareQubitOp declareQubitOp) {
-    declareQubitOp->erase();
+  // erase qubit ops and constant angle ops
+  moduleOp->walk([&](Operation *op) {
+    if (auto castOp = dyn_cast<mlir::quir::DeclareQubitOp>(op))
+      op->erase();
+    else if (auto castOp = dyn_cast<mlir::quir::ConstantOp>(op)) {
```

Comment:
Not sure if this will work but you could try:
```
else if (isa<mlir::quir::ConstantOp>(op) && op->getType().isa<mlir::quit::AngleType>())
```

to avoid the case.